### DISCUSSION TITLE
Console output in CSV format

### DISCUSSION THREAD

**Discussion by lomby92** [symfony/symfony#46632] [Q&A] [answered]
Yesterday I was using the console command `debug:routes` to retrieve the list of endpoints used in my application.
I saw that JSON and MD formats are available but CSV is missing.

Could it be useful to add it? (I could try to contribute)

**Comment by GromNaN** [comment, marked as answer]
Currently none of the debug commands has CSV output format, but why not?
 
> Could it be useful to add it? 

How would it be useful for you?

**Reply by lomby92** [reply]
Example: in case of **debug:routes**, save the output to a `.csv` file and use that file for data visualization (eg: Python script) combining with analytics data of real traffic to recognise route names

**Reply by GromNaN** [reply]
The JSON format would work for that, with the advantages of supporting nested arrays.

**Reply by lomby92** [reply]
In fact, that's what I did. However, this does not mean that the CSV would have been more appropriate due to the tabular nature of the data.

I see that there are other commands with a "list view" output: debug:container, debug:twig, debug:list, debug:form.

Honestly, I'm not experienced in Symfony code, so I don't even know if such a proposal would be a hindrance rather than a help.

**Reply by lyrixx** [reply]
I agree with @GromNaN here. If the goal is to use the data from another (or the same) language JSON really fits better the need.

And if the output should be formatted for a human, the default one is very good. Markdown is nice too.

---

Thus, I think CSV format is not that useful.

But if someone has a legit use case, why not.
